android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -120,12 +120,9 @@ private String appIdFromManifest(ReactApplicationContext context) {
    }
 
    private void removeObservers() {
-      OneSignal.removeEmailSubscriptionObserver(this);
-      OneSignal.removePermissionObserver(this);
-      OneSignal.removeSubscriptionObserver(this);
-      hasSetEmailSubscriptionObserver = false;
-      hasSetPermissionObserver = false;
-      hasSetSubscriptionObserver = false;
+      this.removeEmailSubscriptionObserver();
+      this.removePermissionObserver();
+      this.removeSubscriptionObserver();
```

Comment:
it looks like we're missing the SMS subscription observer removal call here

Reply:
I'll go ahead and take care of it. Cheers

Reply:
thanks Rodrigo! and thanks for the merge.. excited to update to the newest SDK!

Reply:
@frankenthumbs  my pleasure. please let us know if you encounter any problems and we would be happy to help. enjoy!